### PR TITLE
Issue #14631: Updated PARAM_LITERAL of JavadocTokenTypes.java to new format of AST print

### PR DESCRIPTION
**Pull request by SteLeo1602**
Issue: #14631 

**Updated**
image

**Old**
param_literal
Issue: #14631 

**Test.java code used**
```
public class Test {
    /**
     * Javadoc.
     *
     * @param value The parameter of method.
     * @see <a href="https://docs.oracle.com/en/java/">Java Dcoumentation</a>
     */
    public void testFunction(String value) {

    }
}
```
**Command to run**
`java -jar checkstyle-10.14.0-all.jar -J Test.java | sed 's/->/-\&gt;/g; s/\[[0-9]\+:[0-9]\+\]//g'`

**Output**
```
COMPILATION_UNIT -&gt; COMPILATION_UNIT
`--CLASS_DEF -&gt; CLASS_DEF
    |--MODIFIERS -&gt; MODIFIERS
    |   `--LITERAL_PUBLIC -&gt; public
    |--LITERAL_CLASS -&gt; class
    |--IDENT -&gt; Test
    `--OBJBLOCK -&gt; OBJBLOCK
        |--LCURLY -&gt; {
        |--METHOD_DEF -&gt; METHOD_DEF
        |   |--MODIFIERS -&gt; MODIFIERS
        |   |   |--BLOCK_COMMENT_BEGIN -&gt; /*
        |   |   |   |--COMMENT_CONTENT -&gt; *\r\n     * Javadoc.\r\n     *\r\n     * @param value The parameter of method.\r\n     * @see <a href="https://docs.oracle.com/en/java/">Java Dcoumentation</a>\r\n
        |   |   |   |   `--JAVADOC -&gt; JAVADOC
        |   |   |   |       |--NEWLINE -&gt; \r\n
        |   |   |   |       |--LEADING_ASTERISK -&gt;      *
        |   |   |   |       |--TEXT -&gt;  Javadoc.
        |   |   |   |       |--NEWLINE -&gt; \r\n
        |   |   |   |       |--LEADING_ASTERISK -&gt;      *
        |   |   |   |       |--NEWLINE -&gt; \r\n
        |   |   |   |       |--LEADING_ASTERISK -&gt;      *
        |   |   |   |       |--WS -&gt;
        |   |   |   |       |--JAVADOC_TAG -&gt; JAVADOC_TAG
        |   |   |   |       |   |--PARAM_LITERAL -&gt; @param
        |   |   |   |       |   |--WS -&gt;
        |   |   |   |       |   |--PARAMETER_NAME -&gt; value
        |   |   |   |       |   |--WS -&gt;
        |   |   |   |       |   `--DESCRIPTION -&gt; DESCRIPTION
        |   |   |   |       |       |--TEXT -&gt; The parameter of method.
        |   |   |   |       |       `--NEWLINE -&gt; \r\n
        |   |   |   |       |--LEADING_ASTERISK -&gt;      *
        |   |   |   |       |--WS -&gt;
        |   |   |   |       |--JAVADOC_TAG -&gt; JAVADOC_TAG
        |   |   |   |       |   |--SEE_LITERAL -&gt; @see
        |   |   |   |       |   |--WS -&gt;
        |   |   |   |       |   |--HTML_ELEMENT -&gt; HTML_ELEMENT
        |   |   |   |       |   |   `--HTML_TAG -&gt; HTML_TAG
        |   |   |   |       |   |       |--HTML_ELEMENT_START -&gt; HTML_ELEMENT_START
        |   |   |   |       |   |       |   |--START -&gt; <
        |   |   |   |       |   |       |   |--HTML_TAG_NAME -&gt; a
        |   |   |   |       |   |       |   |--WS -&gt;
        |   |   |   |       |   |       |   |--ATTRIBUTE -&gt; ATTRIBUTE
        |   |   |   |       |   |       |   |   |--HTML_TAG_NAME -&gt; href
        |   |   |   |       |   |       |   |   |--EQUALS -&gt; =
        |   |   |   |       |   |       |   |   `--ATTR_VALUE -&gt; "https://docs.oracle.com/en/java/"
        |   |   |   |       |   |       |   `--END -&gt; >
        |   |   |   |       |   |       |--TEXT -&gt; Java Dcoumentation
        |   |   |   |       |   |       `--HTML_ELEMENT_END -&gt; HTML_ELEMENT_END
        |   |   |   |       |   |           |--START -&gt; <
        |   |   |   |       |   |           |--SLASH -&gt; /
        |   |   |   |       |   |           |--HTML_TAG_NAME -&gt; a
        |   |   |   |       |   |           `--END -&gt; >
        |   |   |   |       |   |--NEWLINE -&gt; \r\n
        |   |   |   |       |   `--WS -&gt;
        |   |   |   |       `--EOF -&gt; <EOF>
        |   |   |   `--BLOCK_COMMENT_END -&gt; */
        |   |   `--LITERAL_PUBLIC -&gt; public
        |   |--TYPE -&gt; TYPE
        |   |   `--LITERAL_VOID -&gt; void
        |   |--IDENT -&gt; testFunction
        |   |--LPAREN -&gt; (
        |   |--PARAMETERS -&gt; PARAMETERS
        |   |   `--PARAMETER_DEF -&gt; PARAMETER_DEF
        |   |       |--MODIFIERS -&gt; MODIFIERS
        |   |       |--TYPE -&gt; TYPE
        |   |       |   `--IDENT -&gt; String
        |   |       `--IDENT -&gt; value
        |   |--RPAREN -&gt; )
        |   `--SLIST -&gt; {
        |       `--RCURLY -&gt; }
        `--RCURLY -&gt; }
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -172,13 +172,15 @@ public final class JavadocTokenTypes {
...
-     *       |--PARAM_LITERAL[4x3] : [@param]
-     *       |--WS[4x9] : [ ]
-     *       |--PARAMETER_NAME[4x10] : [T]
-     *       |--WS[4x11] : [ ]
-     *       |--DESCRIPTION[4x12] : [The bar.]
-     *           |--TEXT[4x12] : [The bar.]
+     * JAVADOC_TAG -&gt; JAVADOC_TAG
+     *  |--PARAM_LITERAL -&gt; @param
+     *  |--WS -&gt;
+     *  |--PARAMETER_NAME -&gt; T
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:178]: Please update example to be `@param value The parameter of method.`

Reason: T looks like generic parameter, better to not confuse java engineers.

**Review by romani — CHANGES_REQUESTED** [review]
Same issues as raised in other PR
 plus:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 1 above)
Comment (SteLeo1602) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 2]
```diff
@@ -169,16 +169,16 @@ public final class JavadocTokenTypes {
...
      * <b>Tree:</b>
      * <pre>{@code
-     *   |--JAVADOC_TAG[4x3] : [@param T The bar.]
-     *       |--PARAM_LITERAL[4x3] : [@param]
-     *       |--WS[4x9] : [ ]
-     *       |--PARAMETER_NAME[4x10] : [T]
-     *       |--WS[4x11] : [ ]
-     *       |--DESCRIPTION[4x12] : [The bar.]
-     *           |--TEXT[4x12] : [The bar.]
+     * JAVADOC_TAG ->; JAVADOC_TAG
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:175]: Remove ;

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 2 above)
Comment (SteLeo1602) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**